cloudferrylib/os/actions/vm_snapshot.py

```diff
@@ -48,16 +48,55 @@ def run(self, *args, **kwargs):
         state_to_record = {}
         for vm in self.get_list_of_vms():
             state_to_record.update({vm.id: vm.status})
-        return {self.namespace_variable: state_to_record}
+        return {
+            self.namespace_variable: state_to_record,
+            'rollback_vars': {
+                'vms': []
+            }
+        }
+
+
+class CheckPointVm(action.Action):
```

Comment:
Please add docstring with:
 - Overview
 - Dependent tasks
 - Tasks `CheckPointVm` is dependent on